### PR TITLE
[C API] More tests and nits

### PR DESCRIPTION
**Pull request by taniabogatsch**

### CODE REVIEW

File: src/common/types/value.cpp [hunk 1]
```diff
@@ -1056,6 +1056,9 @@ T Value::GetValueInternal() const {
 	case LogicalTypeId::TIME_TZ:
 		return Cast::Operation<dtime_tz_t, T>(value_.timetz);
 	case LogicalTypeId::TIMESTAMP:
+	case LogicalTypeId::TIMESTAMP_MS:
+	case LogicalTypeId::TIMESTAMP_NS:
+	case LogicalTypeId::TIMESTAMP_SEC:
 	case LogicalTypeId::TIMESTAMP_TZ:
 		return Cast::Operation<timestamp_t, T>(value_.timestamp);
```
[the remark this answers is not in the document]
Comment (taniabogatsch) [reply src/common/types/value.cpp:1063]: Fair - I'll implement these changes in a different PR and then return to this one.

[the remark this answers is not in the document]
Comment (Tishj) [reply src/common/types/value.cpp:1063]: Such as <https://github.com/duckdb/duckdb/blob/1fe66c2ca4d6ae5aa88a46df6b9d4406e19ed35f/src/include/duckdb/common/types/timestamp.hpp#L84> ?

[the remark this answers is not in the document]
Comment (taniabogatsch) [reply src/common/types/value.cpp:1063]: Yes, they already exist, but aren't used in many places. I am adding them to the `Value` `union` and to other timestamp-related functions.

**Comment by jraymakers** [comment]
Coincidentally, I started working on similar functionality a couple weeks ago. I discovered this PR just as I was posting [mine](https://github.com/duckdb/duckdb/pull/14865).

I took a slightly different approach: I added a C API type for each timestamp type, but I didn't change how the C++ layer functions (because that appeared to require more work and seemed more risky).

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**